packages/ra-ui-materialui/src/input/ArrayInput/SimpleFormIterator.tsx

```diff
@@ -111,7 +111,7 @@ export const SimpleFormIterator = (inProps: SimpleFormIteratorProps) => {
                     !Children.only(children).props.source &&
                     // Make sure it's not a FormDataConsumer
                     // @ts-ignore
-                    !Children.only(children).type !== FormDataConsumer
+                    Children.only(children).type !== FormDataConsumer
```

Comment:
Did you try to find a way to remove the `ts-ignore`?

Reply:
no, this is out of scope for this PR IMO